modules/system/assets/js/framework.js

```diff
@@ -30,12 +30,23 @@ if (window.jQuery.request !== undefined) {
         }
 
         /*
-         * Prepare the options and execute the request
+         * Prepare the options
          */
         var $form = options.form ? $(options.form) : $el.closest('form'),
             $triggerEl = !!$form.length ? $form : $el,
             context = { handler: handler, options: options }
 
+        /*
+         * Validate the form client-side
+         */
+        if ($form && $form[0] && !$form[0].checkValidity()) {
+            $form[0].reportValidity();
```

Comment:
@multiwebinc I really want to merge this but IE doesn't support reportValidity(), is it possible for this case to be handled smoother? I mostly don't want a console error message to appear in a browser that doesn't support that method.

Reply:
What would be your preferred method for handling legacy browsers? Just skip the check and send the form like in the current behavior, or implement some sort of custom legacy validation function like they did [here](https://www.sitepoint.com/html5-forms-javascript-constraint-validation-api/), and if so, how would you want the errors to be displayed to the user (e.g. beside the input, or in an alert)? Would the errors need to be translatable as well?

Reply:
@multiwebinc if the polyfill for supporting legacy browsers is simple enough then perhaps we could go that route, but if not then I'd rather just skip the check on the legacy browsers since that's the current behavior and this is more of a progressive enhancement.

Reply:
A polyfill for supporting legacy browsers is not trivial, so I just added a check if the browser supports validation, and if not it behaves as before. Please test because I don't have access to any legacy browsers.

Reply:
Also, I wasn't sure exactly where the function should be placed, so I put it with a few other ones.

Reply:
@w20k are you able to test the polyfill? I don't think it's going to help in this case because IE11 partially supports this API AFAIK, it just doesn't support reportValidity(). Also it doesn't need to be a separate function

Reply:
> it doesn't need to be a separate function

True. I'm just used to breaking long lines up, but I guess that doesn't make much sense this case.

> I don't think it's going to help in this case because IE11 partially supports this API AFAIK

Then we could do `typeof document.createElement( 'input' ).checkValidity == 'function'` [[ref](https://stackoverflow.com/a/8550926)].

Reply:
@multiwebinc it needs to check the `reportValidity()` method, see https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/reportValidity vs https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/checkValidity. IE 10 + supports `checkValidity()` but doesn't support `reportValidity()`

Reply:
@LukeTowers Thanks for the info. Take another look.